src/main/java/com/hazelcast/azure/AzureDiscoveryStrategy.java

```diff
@@ -88,63 +81,103 @@ public void start() {
             String resourceGroup = AzureProperties.getOrNull(AzureProperties.GROUP_NAME, properties);
             String clusterId = AzureProperties.getOrNull(AzureProperties.CLUSTER_ID, properties);
 
-            PagedList<VirtualMachine> virtualMachines = computeManager.virtualMachines().listByResourceGroup(resourceGroup);
-
             ArrayList<DiscoveryNode> nodes = new ArrayList<DiscoveryNode>();
 
-            for (VirtualMachine vm : virtualMachines) {
-                Map<String, String> tags = vm.tags();
-                // a tag is required with the hazelcast clusterid
-                // and the value should be the port number
-                if (tags.get(clusterId) == null) {
-                    continue;
-                }
+            nodes.addAll(discoverVMs(resourceGroup, clusterId));
+            nodes.addAll(discoverScaleSetVMs(resourceGroup, clusterId));
 
-                VirtualMachineInstanceView instanceView = vm.instanceView();
-                // skip any deallocated vms
-                if (!isVirtualMachineOn(instanceView)) {
-                    continue;
-                }
-
-                Integer faultDomainId = instanceView.platformFaultDomain();
-                if (faultDomainId != null) {
-                    memberMetaData.put(PartitionGroupMetaData.PARTITION_GROUP_ZONE, faultDomainId.toString());
-                }
-                int port = Integer.parseInt(tags.get(clusterId));
-                DiscoveryNode node = buildDiscoveredNode(faultDomainId, vm, port);
-
-                if (node != null) {
-                    nodes.add(node);
-                }
-            }
             LOGGER.info("Azure Discovery SPI Discovered " + nodes.size() + " nodes");
             return nodes;
         } catch (Exception e) {
             LOGGER.finest("Failed to discover nodes with Azure SPI", e);
             return null;
         }
+    }
 
+    private SimpleDiscoveryNode createDiscoveryNode(int port, VirtualMachineScaleSetVM vm,
```

Comment:
This code already has `buildDiscoveredNode`, right? Perhaps we should be consistent in the naming?

Reply:
Good point. I will rename it